spec/app/domain/authentication/authn-jwt/fetch_mandatory_claims_to_validate_spec.rb

```diff
@@ -0,0 +1,345 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe('Authentication::AuthnJwt::FetchMandatoryClaimsToValidate') do
+
+  RSpec::Matchers.define :eql_claims_list do |expected|
+    match do |actual|
+      return false unless actual.length == expected.length
+      actual_sorted = actual.sort_by {|obj| obj.name}
+      expected_sorted = expected.sort_by {|obj| obj.name}
+
+      actual_sorted.length.times do |index|
+        return false unless actual_sorted[index].name == expected_sorted[index].name &&
+          actual_sorted[index].value == expected_sorted[index].value
+      end
+
+      return true
+    end
+  end
+
+  let(:all_supported_optional_claims) { %w[iss exp nbf iat].freeze }
+  let(:iss_claim_valid_value) { "iss claim valid value" }
+  let(:token_claim_value) { "value" }
+
+  def mandatory_claims_to_validate_list_with_values(optional_claims)
+    mandatory_claims_to_validate_list = []
+    optional_claims.each do |optional_claim|
+      mandatory_claims_to_validate_list.push(::Authentication::AuthnJwt::JwtMandatoryClaim.new(name: optional_claim, value: claim_value(optional_claim)))
+    end
+
+    mandatory_claims_to_validate_list
+  end
+
+  def claim_value(optional_claim)
+    if optional_claim == 'iss'
```

Comment:
can you document what happens here? why its iss?

Reply:
It is explained in the original class, do you think we need an extra explanation in the UT ?